engine/src/org/pentaho/di/trans/steps/rest/Rest.java

```diff
@@ -302,7 +320,8 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
 
     Object[] r = getRow(); // Get row from input rowset & set row busy!
 
-    if ( r == null ) {
+    logBasic(Arrays.toString(r));
+    if ( r == null && !first ) {
```

Comment:
With a large number of rows with a large number of columns, this would cause a large decrease in performance. Logging information for each row should be done at RowLevel (which is DEBUG level in Log4J), or if it really is more important than other row-level logging, then Detailed or Debug would be fine.

If the user needs to log the row data at Basic level, they can follow the REST step with a Write To Log step and add all fields at level Basic.